pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Package could be either a base or a overlay manifest.
+type Package struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// Labels to add to all objects and selectors.
+	// These labels would also be used to form the selector for apply --prune
+	// Named differently than “labels” to avoid confusion with metadata for this object
+	ObjectLabels map[string]string `json:"objectLabels,omitempty" yaml:"objectLabels,omitempty"`
+
+	// Annotations to add to all objects.
+	ObjectAnnotations map[string]string `json:"objectAnnotations,omitempty" yaml:"objectAnnotations,omitempty"`
+
+	// Partial name that will prefix the name of the base resources.
+	NamePrefix string `json:"namePrefix,omitempty" yaml:"namePrefix,omitempty"`
+
+	// Description of the application package.
+	Description string `json:"description,omitempty" yaml:"description,omitempty"`
+
+	// Search keywords for this application package.
+	Keywords []string `json:"keywords,omitempty" yaml:"keywords,omitempty"`
+
+	// Version of the application package.
+	AppVersion string `json:"appVersion,omitempty" yaml:"appVersion,omitempty"`
```

Comment:
It seems that users would like 2 versions fields: 
From the [Chart.yaml spec](https://github.com/kubernetes/helm/blob/master/docs/charts.md#the-chartyaml-file): 

```
version: A SemVer 2 version (required)
appVersion: The version of the app that this contains (optional). This needn't be SemVer.
```
cc @prydonius

Reply:
@ant31 The standard Kubernetes API metadata already contains a version for the schema of this specification.

Reply:
Yes spec are versioned. 
I was talking about the configuration content. 

e.g User create a first iteration of a 'postgres' package:
```
appVersion: 9.6.0
packageVersion: 1.0.0
bases:
 - deployment.yaml
```

and then later modify the package, to add a configmap
```
appVersion: 9.6.0          
packageVersion: 1.1.0   // upgrade the package version
bases:
 - deployment.yaml
configMaps: 
- type: env
  namePrefix: app-env
  file: app.env
```

Both deploy a postgres 9.6.0 (appVersion), but the configuration iterations would be tracked via the 'packageVersion'  (`version` in case of Chart. )

Helm, app-registry  indexes packages base on the 'packageVersion' and not the 'appVersion', because it's the kubernetes-resources that are deployed and each changes should be tracked.

Reply:
Interesting. We should just start with packageVersion rather than appVersion, then.

The image tag and/or hash is likely to be one of the most frequently customized properties, and any specification in the package metadata would impact both the customization strategy and the update model.

So let's omit appVersion for now.

Reply:
+1 though it's worth mentioning that an appVersion may be dependent on a packageVersion, for example if a new required env var is introduced in an app and the package must be updated to support it.